pkg/envoy/rds/route/route_config.go

```diff
@@ -297,28 +296,22 @@ func buildRetryPolicy(retry *v1alpha1.RetryPolicySpec) *xds_route.RetryPolicy {
 	rp := &xds_route.RetryPolicy{}
 
 	rp.RetryOn = retry.RetryOn
-	rp.NumRetries = &wrapperspb.UInt32Value{
-		Value: uint32(retry.NumRetries),
+	// Remove check when validation webhook is implemented
+	if retry.NumRetries != nil {
+		rp.NumRetries = &wrapperspb.UInt32Value{
+			Value: *retry.NumRetries,
+		}
```

Comment:
Looks like there's a function that can make this marginally easier to read:
https://pkg.go.dev/google.golang.org/protobuf@v1.27.1/types/known/wrapperspb#UInt32
```suggestion
		rp.NumRetries = wrapperspb.UInt32(*retry.NumRetries)
```

Reply:
And if I'm understanding this right, Envoy's default for `num_retries` is 1 (https://www.envoyproxy.io/docs/envoy/v1.19.1/api-v3/config/route/v3/route_components.proto#config-route-v3-retrypolicy) which we were setting explicitly to 0 here before, but now if it's not defined in a retry policy it won't get defined here so Envoy's default is used.

I don't have a strong opinion on what the overall default should be, just pointing out that I think it's changing here.

Reply:
Yes, this change is intended. We want to use Envoy defaults unless we have a specific reason to override the defaults (e.g. circuit breaking where we effectively disable defaults).